client/docs/service_groups.md

```diff
@@ -0,0 +1,160 @@
+be used. That group must be created.
+
+**NOTE:** Services included in service groups must be stored in the
+service endpoint database (see `pdo-service-db` command). Generally
+this means that service endpoint must be added (usually by importing a
+site configuration file) before groups can be created.
+
+## Service Groups Data File ##
+
+Service group information is stored in a database file identified by
+the configuration variable `Service.GroupDatabaseFile` which defaults
+to `${PDO_HOME}/data/groups_db.mdb`. Command line switches allow for
+the use of alternative group files.
+
+## Commands for Working with Service Groups ##
+
+### Service Groups: `pdo-service-groups` ###
+
+The `pdo-service-groups` command provides operations over the collection
+of services in the service groups database.
+
+* `clear` : remove all data from the service database
+* `import` : import service group definitions from a file
+* `export` : export service group definitions to a file
+* `list` : list service groups of a service type
+* `info` : provide detailed information about a group
+
+### Enclave Service Groups
+
+The `pdo-eservice` command provides operations to create, delete and
+modify groups of enclave services. In addition to a list of services,
+`eservice` groups include a preferred service that serves as a default
+location for operating on the contract. If the preferred service is
+`random`, then a different `eservice` will be chosen for each
+operation.
+
+* `create` : create a new service group
+* `create_from_site` : create a new service group using all of the services in a site file
+* `delete` : remove a service group from the database
+* `add` : add services to the service group
+* `remove` : remove services from the service group
+* `set` : replace the current list of services with a new list
+* `use` : select a service for the preferred service
+
+### Provisioning Service Groups
+
+The `pdo-pservice` command provides operations to create, delete and
+modify groups of provisioning services. Provisioning service groups do
+not have any additional policies.
+
+* `create` : create a new service group
+* `create_from_site` : create a new service group using all of the services in a site file
+* `delete` : remove a service group from the database
+* `add` : add services to the service group
+* `remove` : remove services from the service group
+* `set` : replace the current list of services with a new list
+
+### Storage Service Groups
+
+The `pdo-sservice` command provides operations to create, delete and
+modify groups of storage services. Storage service groups have three
+additional policy parameters that may be set: `replicas` specifies the
+minimum number of replicas required for state, `duration` specifies
+the minimum availability duration for each replica, and `persistent`
+identifies a storage service where users can expect to find state
+after the replication duration has expired.
```

Comment:
But do we have any machinery in place where such expectation is somewhat justified? I would have expected that for this to really hold i would have to have correspondingly longer expiration time and/or store it on a third-party storage service?  If not, a user could mistakenly believe setting here a storage to `persistent` it would be guaranteed to be persisted which it wouldn't necessarily?

Reply:
some of this is the preference of the contract creator. however, information is available from every storage service about its expected duration. second, as i said in the document persistence is about expectations, not guarantees. if you want a guarantee of storage then pay for it. this is really (as the document says) about where someone *might* look and expect to find it. note that there does not have to be a persistent service.

Reply:
but so what is the indended  use of that attribute?

Reply:
a hint. a suggestion. an informal agreement. a place where we can go look for state.

Reply:
hmm, still feel that the "peristant" seems to promise more than just the default as in eservice but definitely do not want to fight over it ...